capstone/src/main/resources/static/event.js

```diff
@@ -2,12 +2,14 @@
  * Retrieves events from server
  */
 function getEvents() {
+  showSpinner();
   fetch('user-info').then(response => response.json()).then((data) => {
     if (data.userType != "unknown") {
       getAllEventsForSearch(data);
     } else {
       displayMain(false);
     }
+    hideSpinner();
   });
```

Comment:
Also remove the spinner on error:
```
.catch((error) => {
  // log error
  hideSpinner();
});
```

Example use here: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Uploading_JSON_data